identity-iota-core/src/document/iota_document.rs

```diff
@@ -584,14 +583,14 @@ impl IotaDocument {
   ///
   /// This is the Base58-btc encoded SHA-256 digest of the hex-encoded message id.
   pub fn diff_index(message_id: &MessageId) -> Result<String> {
-    if message_id.is_null() {
+    if message_id_is_null(message_id) {
       return Err(Error::InvalidDocumentMessageId);
     }
 
-    Ok(IotaDID::encode_key(message_id.encode_hex().as_bytes()))
+    Ok(IotaDID::encode_key(message_id.to_string().as_bytes()))
```

Comment:
Looks like I was wrong and we probably need that trait in `identity_iota_core`? It seems a bit dangerous to assume the display implementation for `MessageId` will always to hex either way. Otherwise we might need a comment here?